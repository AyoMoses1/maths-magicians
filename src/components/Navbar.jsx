/* eslint-disable */
import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from '../styles/Navbar.module.css'


const links = [
  { path: '/', text: 'Home' },
  { path: 'calculator', text: 'Calculator' },
  { path: 'quote', text: 'Quote' },
];

const Navbar = () => {
  return (
    <>
      <nav className={styles.navbar}>
        <div className={styles.navlogo}>
          Maths Magicians
        </div>
        <ul className={styles.navlinks}>
          {links.map((link) => (
            <React.Fragment key={link.text}>
              <li>
                <NavLink to={link.path}>{link.text}</NavLink>
              </li>
            </React.Fragment>
          ))}
        </ul>
      </nav>
    </>
  );
};
export default Navbar;
