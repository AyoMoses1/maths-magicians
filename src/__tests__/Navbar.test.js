import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

const MockNavbar = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

it('renders correctly', () => {
  const tree = renderer.create(<MockNavbar />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Navbar tests', () => {
  test('It should contain text content', () => {
    render(<MockNavbar />);
    const textElement = screen.getAllByTestId('Home');
    expect(textElement.length).toBe(1);
  });

  test('It should be visible to the user', () => {
    render(<MockNavbar />);
    const textElement = screen.getAllByRole('list');
    expect(textElement[0]).toBeVisible();
  });
});

describe('tests all nav links', () => {
  test('It Contains Calculator Navlink', () => {
    render(<MockNavbar />);
    const textElement = screen.getAllByTestId('Calculator');
    expect(textElement.length).toBe(1);
  });

  test('It Contains Quote Navlink', () => {
    render(<MockNavbar />);
    const textElement = screen.getAllByTestId('Quote');
    expect(textElement.length).toBe(1);
  });

  test('It Contains Home Navlink', () => {
    render(<MockNavbar />);
    const textElement = screen.getAllByTestId('Home');
    expect(textElement.length).toBe(1);
  });

  test('It Contains Home Navlink', () => {
    render(<MockNavbar />);
    const navlinks = document.querySelectorAll('li a');
    expect(navlinks.length).toBe(3);
  });
});
