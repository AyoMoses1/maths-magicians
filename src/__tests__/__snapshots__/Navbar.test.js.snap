// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="navlogo"
  >
    Maths Magicians
  </div>
  <ul
    className="navlinks"
  >
    <li
      data-testid="Home"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      data-testid="Calculator"
    >
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      data-testid="Quote"
    >
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
